### DISCUSSION TITLE
[General] Failed to validate the transaction. reason: Violated validation rules: Touched unallowed storage slots

### DISCUSSION THREAD

**Discussion by andrewpixelgemium** [zkSync-Community-Hub/zksync-developers#169] [General] [answered]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

How to trace eth_sendRawTransaction error from Json Rpc Api? 

### What is the specific issue or error you're encountering?

I execute the transaction with data like this:
`{
  data: '0x56b3b5200000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000032000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000001e00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000026000000000000000000000000000000000000000000000000000000000000002a00000000000000000000000000000000000000000000000000000000002faf08000000000000000000000000000000000000000000000000000000000652535170000000000000000000000000000000000000000000000000000000065253535000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000067e697000000284f0da45800000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000002faf0800000000000000000000000000000000000000000000000000000000000000001bb11ee31742037c28668fb1c666e01e3f86d23f31a1ae833b60521d9859a4a510000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000b28440600000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000041ff181be01c82f402658a87391e1aef14082862a5e4d0d03c841856a45af6b8d7737125e806b5cbe30d6d184939f1437d1e62e69994c0955f69424f90c3fc64171b00000000000000000000000000000000000000000000000000000000000000',
  to: '0x33fbbC985369192659d9a36a8aD3141335C79c69',
  value: BigNumber { _hex: '0x00', _isBigNumber: true },
  from: '0xB9184F83D293F943154df7F8A7364d09736E7D93',
  chainId: 280,
  nonce: 0,
  type: 113,
  customData: {
    gasPerPubdata: 50000,
    paymasterParams: {
      paymaster: '0x0901866E8dFdd854371251c68dD22D47BE7C164b',
      paymasterInput: '0x949431dc000000000000000000000000d7bfd0bcf23129f1e8d4a3de63e7a59eed6baae8000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000'
    },
    customSignature: Uint8Array(65) [
      102, 184, 150, 137, 157, 142, 137, 158, 122, 141,  26,
      185, 135,  39,  39,  24,  55, 245, 109,  21,  38, 200,
       54,  14, 250,  89, 153,   6,  50, 223, 199,  76,  55,
      107, 148, 200, 126, 140, 230, 201, 122, 174, 169,  18,
       89, 115, 191, 113,  51,  71, 133, 178,  45,  48, 250,
      201,  61, 236,   8, 198, 137, 169,  68,  62,  28
    ]
  },
  gasPrice: BigNumber { _hex: '0x0ee6b280', _isBigNumber: true },
  gasLimit: BigNumber { _hex: '0x0c29c8', _isBigNumber: true }
}`

And i met the error: 

`failed to validate the transaction. reason: Violated validation rules: Touched unallowed storage slots: address d7bfd0bcf23129f1e8d4a3de63e7a59eed6baae8, key: 360894a13ba1a3210667c828492db98dca3e2076cc3735a920a3ca505d382bbc`

with the requestBody:

`requestBody="{\"method\":\"eth_sendRawTransaction\",\"params\":[\"0x71f904cd80840ee6b280840ee6b280830c29c89433fbbc985369192659d9a36a8ad3141335c79c6980b903a456b3b5200000000000000000000000000000000000000000000000000000000000000040000000000000000000000000000000000000000000000000000000000000032000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000016000000000000000000000000000000000000000000000000000000000000001a000000000000000000000000000000000000000000000000000000000000001e00000000000000000000000000000000000000000000000000000000000000220000000000000000000000000000000000000000000000000000000000000026000000000000000000000000000000000000000000000000000000000000002a00000000000000000000000000000000000000000000000000000000002faf08000000000000000000000000000000000000000000000000000000000652535170000000000000000000000000000000000000000000000000000000065253535000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000067e697000000284f0da45800000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000002faf0800000000000000000000000000000000000000000000000000000000000000001bb11ee31742037c28668fb1c666e01e3f86d23f31a1ae833b60521d9859a4a510000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000b28440600000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000041ff181be01c82f402658a87391e1aef14082862a5e4d0d03c841856a45af6b8d7737125e806b5cbe30d6d184939f1437d1e62e69994c0955f69424f90c3fc64171b00000000000000000000000000000000000000000000000000000000000000820118808082011894b9184f83d293f943154df7f8a7364d09736e7d9382c350c0b84166b896899d8e899e7a8d1ab98727271837f56d1526c8360efa59990632dfc74c376b94c87e8ce6c97aaea9125973bf71334785b22d30fac93dec08c689a9443e1cf89b940901866e8dfdd854371251c68dd22d47be7c164bb884949431dc000000000000000000000000d7bfd0bcf23129f1e8d4a3de63e7a59eed6baae8000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000600000000000000000000000000000000000000000000000000000000000000000\"],\"id\":59,\"jsonrpc\":\"2.0\"}"`

So how can i trace the eth_sendRawTransaction error from Json Rpc Api?

### Can you share the error messages or logs you're receiving, if any?

Above

### Have you made any recent changes to the contract before encountering this issue?

No

### Are there any external libraries or contracts that your contract interacts with?

No

### Can you provide the relevant portions of your contract code where the issue is occurring?

No

### Have you tried to isolate the problem, and if so, what were the results?

No

### What steps have you already taken to try to resolve the issue?

No

**Comment by Romsters** [comment, marked as answer]
Hi @andrewpixelgoal3, the reason is the account logic can only touch slots that belong to the account. Please check https://era.zksync.io/docs/reference/concepts/account-abstraction.html#limitations-of-the-verification-step

**Comment by andybrierton** [comment]
To trace the error from eth_sendRawTransaction in the JSON RPC API, you need to analyze the error message returned by the API. In your case, the error message indicates that the transaction failed due to "Touched unallowed storage slots." This suggests that the transaction attempted to access storage slots that are not permitted.

To resolve this issue, you should review your transaction data and ensure that it complies with the rules and constraints of the contract you are interacting with. Specifically, check if the contract restricts access to certain storage slots or imposes other validation rules that your transaction violates.

Additionally, consider debugging your transaction data to identify any discrepancies or errors. You can log intermediate values, double-check the contract ABI and function parameters, and verify that the transaction data is properly formatted and encoded.

If you continue to encounter issues, you may need to consult the documentation or source code of the contract to understand its behavior and requirements better. Alternatively, you can seek assistance from the contract developers or the community for further guidance on resolving the issue.

**Comment by clevelandmarcus** [comment]
To trace the eth_sendRawTransaction error from the JSON RPC API, you can examine the error message returned by the API call. Look for details such as the reason for the failure and any specific error codes or messages provided. This information can help identify the cause of the error and guide you in troubleshooting and resolving the issue.

**Comment by sundaramjacob** [comment]
To trace the "eth_sendRawTransaction" error from the JSON RPC API, you can analyze the transaction data and the error message. Specifically, check for any violations of validation rules, such as touched unallowed storage slots or other constraints. Ensure that the transaction data and parameters are correctly formatted and comply with the contract's requirements. You may also need to debug the contract code to identify any issues causing the error.
